alf/utils/checkpoint_utils.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import warnings
+import torch
+from absl import logging
+
+from alf.algorithms.algorithm import Algorithm
+
+
+class Checkpointer(object):
+    """A checkpoint manager for saving and loading checkpoints."""
+
+    def __init__(self, ckpt_dir, **kwargs):
+        """A class for making checkpoints.
+
+        Args:
+            ckpt_dir: The directory to save checkpoints. Create ckpt_dir if
+                it doesn't exist.
+            kwargs: Items to be included in the checkpoint. Each item needs
+                to have state_dict and load_state_dict implemented.
+                For instance of Algorithm, only the root need to be passed in,
+                all the children modules and optimizers are automatically
+                extracted and checkpointed.
```

Comment:
Comment on what happens if children modules are also passed in

Reply:
added